apiAuditLogs/apiAuditLogs.go

```diff
@@ -0,0 +1,277 @@
+		"getTeamsUnreadForUser":                true,
+		"getTeamUnread":                        true,
+		"getTotalUsersStats":                   true,
+		"getUser":                              true,
+		"getUserAccessToken":                   true,
+		"getUserAccessTokens":                  true,
+		"getUserAccessTokensForUser":           true,
+		"getUserByEmail":                       true,
+		"getUserByUsername":                    true,
+		"getUsers":                             true,
+		"getUsersByGroupChannelIds":            true,
+		"getUsersByIds":                        true,
+		"getUsersByNames":                      true,
+		"getUserStatus":                        true,
+		"getUserStatusesByIds":                 true,
+		"getUserTermsOfService":                true,
+		"getWebappPlugins":                     true,
+		"listAutocompleteCommands":             true,
+		"listCommands":                         true,
+		"openDialog":                           true,
+		"patchChannelModerations":              true,
+		"pinPost":                              true,
+		"pushNotificationAck":                  true,
+		"saveReaction":                         true,
+		"searchAllChannels":                    true,
+		"searchArchivedChannelsForTeam":        true,
+		"searchChannelsForTeam":                true,
+		"searchEmojis":                         true,
+		"searchGroupChannels":                  true,
+		"searchPosts":                          true,
+		"searchTeams":                          true,
+		"searchUserAccessTokens":               true,
+		"searchUsers":                          true,
+		"setPostUnread":                        true,
+		"submitDialog":                         true,
+		"teamExists":                           true,
+		"teamMembersMinusGroupMembers":         true,
+		"testElasticsearch":                    true,
+		"testEmail":                            true,
+		"testLdap":                             true,
+		"testS3":                               true,
+		"testSiteURL":                          true,
+		"unpinPost":                            true,
+		"updateUserStatus":                     true,
+		"uploadFileStream":                     true,
+		"viewChannel":                          true,
+	}
+
+	if pass.Pkg.Path() != "github.com/mattermost/mattermost-server/v5/api4" {
+		return nil, nil
+	}
+
+	inspect := pass.ResultOf[inspect.Analyzer].(*inspector.Inspector)
+	nodeFilter := []ast.Node{
+		(*ast.FuncDecl)(nil),
+	}
+	inspect.Preorder(nodeFilter, func(n ast.Node) {
+		funDecl := n.(*ast.FuncDecl)
+
+		if strings.HasSuffix(pass.Fset.File(n.Pos()).Name(), "_test.go") {
+			return
+		}
+		if strings.HasSuffix(pass.Fset.File(n.Pos()).Name(), "apitestlib.go") {
+			return
+		}
+		if obj, ok := pass.TypesInfo.Defs[funDecl.Name].(*types.Func); ok {
+			var fact facts.IsApiHandler
+			if !pass.ImportObjectFact(obj, &fact) {
+				return
+			}
+		}
+		if whiteList[funDecl.Name.Name] {
```

Comment:
Let's move this above so that we avoid the fact check unless needed.

Reply:
done